src/common/iop_order.c

```diff
@@ -1411,6 +1413,144 @@ static void _ioppr_check_rules(GList *iop_list, const int imgid, const char *msg
   if(fences) g_list_free(fences);
 }
 
+
+// how is on-the-fly conversion done
+// Currently a hack to support v3 history to later
+void dt_ioppr_convert_onthefly(const int imgid)
+{
+  int my_iop_order_version = 0;
+
+  sqlite3_stmt *stmt;
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), "SELECT iop_order_version FROM main.images WHERE id = ?1",
+                              -1, &stmt, NULL);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, imgid);
+  if(sqlite3_step(stmt) == SQLITE_ROW)
+  {
+    my_iop_order_version = sqlite3_column_int(stmt, 0);
+  }
+  sqlite3_finalize(stmt);
+
+  if (my_iop_order_version == DT_IOP_ORDER_VERSION) return;
+
+  if (my_iop_order_version != 3) return; // this keeps other edit as they are
+
+  // ************** from here on we deal only with the v3 history problems; although *******************************
+
+  // As we have to calculate within the images history data we will create a struct array holding all relevant data
+  // for housekeeping.
+  // Also this ensures we can write test output before we really write data to main.history
+
+  int history_size = 0;
+  // We need the size of the images history for the struct array
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
+                              "SELECT COUNT(*) FROM main.history WHERE imgid = ?1", -1, &stmt, NULL);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, imgid);
+  if(sqlite3_step(stmt) == SQLITE_ROW) history_size = sqlite3_column_int(stmt, 0);
+  sqlite3_finalize(stmt);
+
+  if (history_size <1)
+  {
+    fprintf(stderr,"\n[dt_ioppr_convert_onthefly] for image %i has no valid history\n",imgid);
+    return;
+  }
+
+  GList *current_iop_list = dt_ioppr_get_iop_order_list(NULL);
+
+  // get the number of known iops
+  int valid_iops = 0;
+
+  GList *iops_order = g_list_last(current_iop_list);
+  while(iops_order)
+  {
+    valid_iops++; 
+    iops_order = g_list_previous(iops_order);
+  }
```

Comment:
Equivalent to:
`const int valid_iops = g_list_length (current_iop_list);`

Reply:
Thanks :-)